{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## DALL-E2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "from openai import OpenAI\n",
    "client = OpenAI()\n",
    "\n",
    "response = client.images.edit(\n",
    "    model=\"dall-e-2\",\n",
    "    image=open(\"../imgs/before.png\", 'rb'),\n",
    "    prompt=\"Plant a ginkgo tree along the roadway\",\n",
    "    n=1,\n",
    "    size=\"1024x1024\"\n",
    ")\n",
    "\n",
    "image_url = response.data[0].url"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "https://oaidalleapiprodscus.blob.core.windows.net/private/org-aJQkCxMN3ziCmWhz0Cfkr2gC/user-pfz8ieGIw5rDRyJoXPGRX0Xw/img-s2xvgFWVNjPrWDRTlRXnDJGe.png?st=2024-07-12T07%3A45%3A18Z&se=2024-07-12T09%3A45%3A18Z&sp=r&sv=2023-11-03&sr=b&rscd=inline&rsct=image/png&skoid=6aaadede-4fb3-4698-a8f6-684d7786b067&sktid=a48cca56-e6da-484e-a814-9c849652bcb3&skt=2024-07-11T23%3A22%3A45Z&ske=2024-07-12T23%3A22%3A45Z&sks=b&skv=2023-11-03&sig=a1ayLBdd5tC8bwAOdPmJnJgg2G64DdrAxXT%2Bg8i7tHU%3D\n"
     ]
    }
   ],
   "source": [
    "print(image_url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gooey.AI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "ename": "AssertionError",
     "evalue": "b'{\"detail\":[{\"loc\":[\"body\",\"input_image\"],\"msg\":\"invalid or missing URL scheme. Please make sure the URL is correct and accessible. If you are trying to use a local file, please use the [Upload Files via Form Data] option on https://gooey.ai/api/ to upload the file directly.\",\"type\":\"value_error.url.scheme\"}]}'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[5], line 16\u001b[0m\n\u001b[0;32m      4\u001b[0m payload \u001b[38;5;241m=\u001b[39m {\n\u001b[0;32m      5\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124minput_image\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m../imgs/before.png\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m      6\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtext_prompt\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mPlant street trees along the sidewalk\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m      7\u001b[0m }\n\u001b[0;32m      9\u001b[0m response \u001b[38;5;241m=\u001b[39m requests\u001b[38;5;241m.\u001b[39mpost(\n\u001b[0;32m     10\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhttps://api.gooey.ai/v2/Img2Img/\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m     11\u001b[0m     headers\u001b[38;5;241m=\u001b[39m{\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     14\u001b[0m     json\u001b[38;5;241m=\u001b[39mpayload,\n\u001b[0;32m     15\u001b[0m )\n\u001b[1;32m---> 16\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m response\u001b[38;5;241m.\u001b[39mok, response\u001b[38;5;241m.\u001b[39mcontent\n\u001b[0;32m     18\u001b[0m result \u001b[38;5;241m=\u001b[39m response\u001b[38;5;241m.\u001b[39mjson()\n\u001b[0;32m     19\u001b[0m \u001b[38;5;28mprint\u001b[39m(response\u001b[38;5;241m.\u001b[39mstatus_code, result)\n",
      "\u001b[1;31mAssertionError\u001b[0m: b'{\"detail\":[{\"loc\":[\"body\",\"input_image\"],\"msg\":\"invalid or missing URL scheme. Please make sure the URL is correct and accessible. If you are trying to use a local file, please use the [Upload Files via Form Data] option on https://gooey.ai/api/ to upload the file directly.\",\"type\":\"value_error.url.scheme\"}]}'"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import requests\n",
    "\n",
    "payload = {\n",
    "    \"input_image\": \"../imgs/before.png\",\n",
    "    \"text_prompt\": \"Plant street trees along the sidewalk\",\n",
    "}\n",
    "\n",
    "response = requests.post(\n",
    "    \"https://api.gooey.ai/v2/Img2Img/\",\n",
    "    headers={\n",
    "        \"Authorization\": \"Bearer \" + os.environ[\"GOOEY_API_KEY\"],\n",
    "    },\n",
    "    json=payload,\n",
    ")\n",
    "assert response.ok, response.content\n",
    "\n",
    "result = response.json()\n",
    "print(response.status_code, result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "AssertionError",
     "evalue": "b'{\"detail\":{\"id\":\"i0y5m0ikntzb\",\"url\":\"https://gooey.ai/Lipsync/?run_id=i0y5m0ikntzb&uid=g9wNqCJzVeQeh83Env4eJ4lPd9j2\",\"created_at\":\"2024-07-15T04:34:14.722398+00:00\",\"error\":\"It seems like you are not using a human face for the input. This AI tool requires a human face to run the lipsync workflow. Know more [here](https://gooey.ai/docs/guides/lip-sync-animation-generator#id-3akkpf7ao60t). \\\\n\\\\nDetails:\\\\n\\\\nError in GPU Task gooey-gpu/wav2lip_gan.pth:wav2lip - Command \\'[\\'ffmpeg\\', \\'-y\\', \\'-stream_loop\\', \\'-1\\', \\'-i\\', \\'/tmp/tmp4zxnb0hi/face.png\\', \\'-i\\', \\'/tmp/tmp4zxnb0hi/audio.wav\\', \\'-shortest\\', \\'-fflags\\', \\'+shortest\\', \\'-max_interleave_delta\\', \\'100M\\', \\'-map\\', \\'0:v:0\\', \\'-map\\', \\'1:a:0\\', \\'-pix_fmt\\', \\'yuv420p\\', \\'/tmp/tmp4zxnb0hi/result_voice.mp4\\']\\' returned non-zero exit status 1.\\\\n\"}}'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[3], line 18\u001b[0m\n\u001b[0;32m      8\u001b[0m payload \u001b[38;5;241m=\u001b[39m {}\n\u001b[0;32m     10\u001b[0m response \u001b[38;5;241m=\u001b[39m requests\u001b[38;5;241m.\u001b[39mpost(\n\u001b[0;32m     11\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhttps://api.gooey.ai/v2/Lipsync/form/?run_id=fecsii61rs6e&uid=fm165fOmucZlpa5YHupPBdcvDR02\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m     12\u001b[0m     headers\u001b[38;5;241m=\u001b[39m{\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     16\u001b[0m     data\u001b[38;5;241m=\u001b[39m{\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mjson\u001b[39m\u001b[38;5;124m\"\u001b[39m: json\u001b[38;5;241m.\u001b[39mdumps(payload)},\n\u001b[0;32m     17\u001b[0m )\n\u001b[1;32m---> 18\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m response\u001b[38;5;241m.\u001b[39mok, response\u001b[38;5;241m.\u001b[39mcontent\n\u001b[0;32m     20\u001b[0m result \u001b[38;5;241m=\u001b[39m response\u001b[38;5;241m.\u001b[39mjson();\n\u001b[0;32m     21\u001b[0m \u001b[38;5;28mprint\u001b[39m(response\u001b[38;5;241m.\u001b[39mstatus_code, result)\n",
      "\u001b[1;31mAssertionError\u001b[0m: b'{\"detail\":{\"id\":\"i0y5m0ikntzb\",\"url\":\"https://gooey.ai/Lipsync/?run_id=i0y5m0ikntzb&uid=g9wNqCJzVeQeh83Env4eJ4lPd9j2\",\"created_at\":\"2024-07-15T04:34:14.722398+00:00\",\"error\":\"It seems like you are not using a human face for the input. This AI tool requires a human face to run the lipsync workflow. Know more [here](https://gooey.ai/docs/guides/lip-sync-animation-generator#id-3akkpf7ao60t). \\\\n\\\\nDetails:\\\\n\\\\nError in GPU Task gooey-gpu/wav2lip_gan.pth:wav2lip - Command \\'[\\'ffmpeg\\', \\'-y\\', \\'-stream_loop\\', \\'-1\\', \\'-i\\', \\'/tmp/tmp4zxnb0hi/face.png\\', \\'-i\\', \\'/tmp/tmp4zxnb0hi/audio.wav\\', \\'-shortest\\', \\'-fflags\\', \\'+shortest\\', \\'-max_interleave_delta\\', \\'100M\\', \\'-map\\', \\'0:v:0\\', \\'-map\\', \\'1:a:0\\', \\'-pix_fmt\\', \\'yuv420p\\', \\'/tmp/tmp4zxnb0hi/result_voice.mp4\\']\\' returned non-zero exit status 1.\\\\n\"}}'"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import requests\n",
    "import json\n",
    "\n",
    "files = [\n",
    "    (\"input_face\", open(\"../imgs/before.png\", \"rb\"))\n",
    "]\n",
    "payload = {}\n",
    "\n",
    "response = requests.post(\n",
    "    \"https://api.gooey.ai/v2/Lipsync/form/?run_id=fecsii61rs6e&uid=fm165fOmucZlpa5YHupPBdcvDR02\",\n",
    "    headers={\n",
    "        \"Authorization\": \"Bearer \" + os.environ[\"GOOEY_API_KEY\"],\n",
    "    },\n",
    "    files=files,\n",
    "    data={\"json\": json.dumps(payload)},\n",
    ")\n",
    "assert response.ok, response.content\n",
    "\n",
    "result = response.json();\n",
    "print(response.status_code, result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## playground"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'replica'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[10], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mreplica\u001b[39;00m\n",
      "\u001b[1;31mModuleNotFoundError\u001b[0m: No module named 'replica'"
     ]
    }
   ],
   "source": [
    "import replicate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "image = open(\"../imgs/before.png\", \"rb\")\n",
    "\n",
    "input = {\n",
    "    \"prompt\": \"Plant street trees along the sidewalk\",\n",
    "    \"image\": image\n",
    "}\n",
    "\n",
    "output = replicate.run(\n",
    "    \"jyoung105/playground-v2.5:e9f13772a98366f2dc1cab993da27b5676a7d3cd4fc8d06753f09397a8cc01b9\",\n",
    "    input=input\n",
    ")\n",
    "print(output)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "qa_system",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
